cdap-api-common/src/main/java/co/cask/cdap/api/data/schema/Schema.java

```diff
@@ -99,21 +99,53 @@ public boolean isSimpleType() {
    *
    */
   public enum LogicalType {
-    DATE(Type.INT),
-    TIMESTAMP_MILLIS(Type.LONG),
-    TIMESTAMP_MICROS(Type.LONG),
-    TIME_MILLIS(Type.INT),
-    TIME_MICROS(Type.LONG);
+    DATE(Type.INT, "date"),
+    TIMESTAMP_MILLIS(Type.LONG, "timestamp-millis"),
+    TIMESTAMP_MICROS(Type.LONG, "timestamp-micros"),
+    TIME_MILLIS(Type.INT, "time-millis"),
+    TIME_MICROS(Type.LONG, "time-micros");
 
     private final Type type;
+    private final String name;
+
+    private static final Map<String, LogicalType> LOOKUP_BY_NAME;
+    static {
+      LOOKUP_BY_NAME = new HashMap<>();
+      for (LogicalType logicalType: values()) {
+        LOOKUP_BY_NAME.put(logicalType.name, logicalType);
+      }
+    }
 
     /**
      * Creates {@link LogicalType} with underlying primitive type.
      *
      * @param type primitive type on which this {@link LogicalType} relies on
+     * @param name name of the {@link LogicalType}
      */
-    LogicalType(Type type) {
+    LogicalType(Type type, String name) {
       this.type = type;
+      this.name = name;
+    }
+
+    /**
+     * @return returns the name of this logical type.
+     */
+    public String getName() {
```

Comment:
should call it something other than name, as it is confusing to have `.name()` and `.getName()`. How about 'token'?

Reply:
yea I agree, `name` is a confusing variable name. I could not think of anything better. `token` seems fine to me.